src/profile/ResetPassword.js

```diff
@@ -1,148 +1,251 @@
+            autoFocus
+            label={I18n.get("email")}
+            {...register("email", {
+              required: "email is required",
+              pattern: {
+                value: /^[^\s@]+@([^\s@.,]+\.)+[^\s@.,]{2,}$/,
+                message: "invalid email address",
+              },
+            })}
+          />
+        </div>
+        <span className="error">{errors.email && errors.email.message}</span>
+        <div>
+          <LoaderButton
+            loadingText={I18n.get("sending")} //
+            text={I18n.get("sendConfirmation")} //
+            isLoading={isSendingCode} //
+            disabled={disabled} //
+            type="submit"
+            color="primary"
+            variant="contained"
+          />
+        </div>
+      </form>
+    </div>
   );
 
   const renderConfirmationForm = () => (
-    <form onSubmit={handleConfirmClick}>
-      <FormGroup bsSize="large" controlId="code">
-        <ControlLabel>{I18n.get("confirmationCode")}</ControlLabel>
-        <FormControl
-          autoFocus
-          type="tel"
-          value={state.code}
-          onChange={handleChange}
-        />
-        <HelpBlock>{I18n.get("checkEmail")}</HelpBlock>
-      </FormGroup>
-      <hr />
-      <FormGroup bsSize="large" controlId="password">
-        <ControlLabel>{I18n.get("newPassword")}</ControlLabel>
-        <FormControl
-          type="password"
-          value={state.password}
-          onChange={handleChange}
-        />
-      </FormGroup>
-      <FormGroup bsSize="large" controlId="confirmPassword">
-        <ControlLabel>{I18n.get("confirm")}</ControlLabel>
-        <FormControl
-          type="password"
-          onChange={handleChange}
-          value={state.confirmPassword}
-        />
-      </FormGroup>
-      <LoaderButton
-        block
-        type="submit"
-        size="large"
-        text={I18n.get("confirm")}
-        loadingText={I18n.get("confirming")}
-        isLoading={state.isConfirming}
-        disabled={!validateResetForm()}
-      />
-    </form>
+    <div className="Form">
+      <Container>
+        <form
+          className={classes.root}
+          onSubmit={handleSubmit2(handleConfirmClick)}
+        >
+          <div>
+            <TextField
+              id="code"
+              variant="filled"
+              size="medium"
+              autoFocus
+              label={I18n.get("confirmationCode")}
+              type="tel"
+              {...register2("code", {
+                required: "code is required",
+                minLength: {
+                  value: 6,
+                  message: "codes are 6 digits long",
+                },
+                maxLength: {
+                  value: 6,
+                  message: "codes are 6 digits long",
+                },
+              })}
+            />
+          </div>
+
+          <span className="error">{errors2.code && errors2.code.message}</span>
+          <hr />
+          <div>
```

Comment:
unnecessary div

Reply:
When I take these divs out, the password and confirm password TextFields join together which maybe is a bit untidy. I think you can achieve the same with the Grid container and Grid item tags, but I am not sure if that would be slower than the divs. Is there another way to do this or would just taking the divs out be okay anyway?
<img width="320" alt="password" src="https://user-images.githubusercontent.com/60603397/164774380-0d6b47b2-fd1a-4a71-8d1f-ccd015563e52.png">

Reply:
I think this is because the break in the code happens after the error message which only gets rendered when an invalid password gets entered.

Reply:
On second look this layout seems fine, so I have removed the div.